instrumentation/akka/akka-http-10.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/akkahttp/client/HttpExtClientInstrumentation.java

```diff
@@ -43,55 +46,81 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class SingleRequestAdvice {
 
+    public static class AdviceScope {
+      private final Context context;
+      private final Scope scope;
+      private final HttpRequest request;
+
+      private AdviceScope(Context context, Scope scope, HttpRequest request) {
+        this.context = context;
+        this.scope = scope;
+        this.request = request;
+      }
+
+      public static Object[] start(HttpRequest request) {
+        Context parentContext = currentContext();
+        if (!instrumenter().shouldStart(parentContext, request)) {
+          return new Object[] {null, request};
+        }
+        Context context = instrumenter().start(parentContext, request);
+        // Making context current is required for header context propagation to work as expected
+        // because it implicitly relies on the current context when injecting headers.
+        Scope scope = context.makeCurrent();
+        // Request is immutable, so we have to assign new value once we update headers
+        HttpRequest modifiedRequest = setter().inject(request);
```

Comment:
[for reviewer] Here ordering matters, inlining the `scope` variable in the call to `AdviceScope` constructor call below breaks context propagation, hence the added comment to make that slightly more obvious.